ts/sagas/startup/watchApplicationActivitySaga.ts

```diff
@@ -1,106 +1,57 @@
+import { Effect, Task } from "redux-saga";
+import { call, cancel, fork, put, takeEvery } from "redux-saga/effects";
 import { ActionType, getType } from "typesafe-actions";
 
 import { backgroundActivityTimeout } from "../../config";
-import AppNavigator from "../../navigation/AppNavigator";
 import {
   applicationChangeState,
-  ApplicationState,
-  startApplicationInitialization
+  ApplicationState
 } from "../../store/actions/application";
-import {
-  navigateToBackgroundScreen,
-  navigateToMessageDetailScreenAction
-} from "../../store/actions/navigation";
-import { navigationHistoryPush } from "../../store/actions/navigationHistory";
-import { clearNotificationPendingMessage } from "../../store/actions/notifications";
-import { navigationStateSelector } from "../../store/reducers/navigation";
-import {
-  PendingMessageState,
-  pendingMessageStateSelector
-} from "../../store/reducers/notifications/pendingMessage";
-import { GlobalState } from "../../store/reducers/types";
-import { saveNavigationStateSaga } from "../startup/saveNavigationStateSaga";
+import { identificationRequest } from "../../store/actions/identification";
+import { startTimer } from "../../utils/timer";
 
 /**
- * Listen to APP_STATE_CHANGE_ACTION and if needed force the user to provide
- * the PIN
+ * Listen to APP_STATE_CHANGE_ACTION and if needed force the user to identify
  */
-// tslint:disable-next-line:cognitive-complexity
 export function* watchApplicationActivitySaga(): IterableIterator<Effect> {
   const backgroundActivityTimeoutMillis = backgroundActivityTimeout * 1000;
 
   // tslint:disable-next-line:no-let
   let lastState: ApplicationState = "active";
 
   // tslint:disable-next-line:no-let
-  let lastUpdateAtMillis: number | undefined;
+  let identificationBackgroundTimer: Task | undefined;
 
   yield takeEvery(getType(applicationChangeState), function*(
     action: ActionType<typeof applicationChangeState>
   ) {
-    // listen for changes in application state
+    // Listen for changes in application state
     const newApplicationState: ApplicationState = action.payload;
 
-    // get the time elapsed from the last change in state
-    const nowMillis = new Date().getTime();
-    const timeElapsedMillis = lastUpdateAtMillis
-      ? nowMillis - lastUpdateAtMillis
-      : nowMillis;
-
     if (lastState !== "background" && newApplicationState === "background") {
-      // The app is going into background
-      // Save the navigation state so we can restore it later when the app come
-      // back to the active state
-      yield call(saveNavigationStateSaga);
-
-      // Make sure that when the app come back active, the BackgrounScreen
-      // gets loaded first
-      // FIXME: not that this creates a quick blue flash in case after restoring
-      //        the app we don't ask a PIN
-      yield put(navigateToBackgroundScreen);
+      // Start the background timer
+      identificationBackgroundTimer = yield fork(
```

Comment:
it doesn't look like `startIdentificationBackgroundTimer` returns anything

Reply:
The return of "fork" effect is a redux-saga Task you can cancel.